### PR TITLE
RavenDB-12319 spawn processes without handle inheritance to avoid getting HttpRequestException because of closed handle

### PR DESCRIPTION
**Pull request by ppekrol**
This is still WIP, I want CI to start testing this

### CODE REVIEW

File: src/Sparrow/RavenProcess.cs [hunk 1]
```diff
@@ -0,0 +1,649 @@
...
+        {
+            if (startInfo.UseShellExecute)
+                throw new NotSupportedException();
+
+            if (PlatformDetails.RunningOnPosix)
+                return new RavenProcess(Process.Start(startInfo));
+
+            var process = new RavenProcess();
+            if (process.StartWithCreateProcess(startInfo) == false)
+                throw new InvalidOperationException();
```
Comment (ayende) [inline src/Sparrow/RavenProcess.cs]: Need error handling here.

File: src/Sparrow/RavenProcess.cs [hunk 2]
```diff
@@ -0,0 +1,649 @@
...
+
+            _processHandle?.Dispose();
+            _standardOutput?.Dispose();
+            _standardInput?.Dispose();
+            _standardError?.Dispose();
+        }
+
+        private unsafe bool StartWithCreateProcess(ProcessStartInfo startInfo)
+        {
+            var commandLine = BuildCommandLine(startInfo.FileName, startInfo.Arguments);
```
Comment (ayende) [inline src/Sparrow/RavenProcess.cs]: How much of this is Windows only?

Comment (ppekrol) [reply src/Sparrow/RavenProcess.cs]: All

**Comment by ppekrol** [comment]
@ayende I've compiled new System.Diagnostics.Process.dll (for Windows only) with InheritHandles option.

**Comment by ppekrol** [comment]
@ayende I will merge this on Monday, after release

**Pull request merged**